### ISSUE TITLE
subset maf based on clinical feature

### ISSUE THREAD

**Issue by patidarr** [PoisonAlien/maftools#118]
Hi,
I have a massive maf and would like to filter it based on clinical features(e.g. Age), is it possible?

Thanks!!

**Comment by PoisonAlien** [comment]
Hi again,
You can subset however you like. For example using toy laml dataset bundled with maftools if I wanted to make MAF file of only M2 subtype of disease..

```r
>cd = getClinicalData(x = laml)
#This is what clinical data contains
> head(cd)
   Tumor_Sample_Barcode FAB_classification days_to_last_followup Overall_Survival_Status
1:         TCGA-AB-2802                 M4                   365                       1
2:         TCGA-AB-2803                 M3                   792                       1
3:         TCGA-AB-2804                 M3                  2557                       0
4:         TCGA-AB-2805                 M0                   577                       1
5:         TCGA-AB-2806                 M1                   945                       1
6:         TCGA-AB-2807                 M1                   181                       1

#First fetch all samples with M2 subtype (In your case you need to get sample with Age (e.g, Age > 50)
>m2.samples = cd[FAB_classification %in% 'M2', Tumor_Sample_Barcode]

#Once you have your sample list, use subsetMaf
>m2.maf = subsetMaf(maf = laml, tsb = m2.samples, mafObj = TRUE)
> m2.maf
An object of class  MAF 
                   ID          summary  Mean Median
 1:        NCBI_Build               37    NA     NA
 2:            Center genome.wustl.edu    NA     NA
 3:           Samples               44    NA     NA
 4:            nGenes              353    NA     NA
 5:   Frame_Shift_Del               18 0.409      0
 6:   Frame_Shift_Ins               20 0.455      0
 7:      In_Frame_Del                2 0.045      0
 8:      In_Frame_Ins               11 0.250      0
 9: Missense_Mutation              332 7.545      8
10: Nonsense_Mutation               30 0.682      0
11:       Splice_Site               25 0.568      0
12:             total              438 9.955     10
```

tl;dr : You need to get sample names of interest and subset. Hope this was clear.

**Comment by patidarr** [comment]
Awesome, Thanks!!
